### ISSUE TITLE
CSharpScript does not work with .net Core 1.1.0

### ISSUE THREAD

**Issue by kukjevov** [dotnet/roslyn#15352]
**Version Used**: 
"Microsoft.CodeAnalysis.CSharp.Scripting": "2.0.0-rc"

project target type
```json
"netstandard1.6":
{
    "imports":
    [
        "dnxcore50",
        "portable-net45+win8"
    ],
    "dependencies":
    {
        "NETStandard.Library": "1.6.1"
    }
}
```

**Steps to Reproduce**:

I have following class:
```csharp
public class TestClass
{
    public int X
    {
        get;
        set;
    }

    public int Y
    {
        get;
        set;
    }
}
```

And following code always crash
```csharp
var cls = new TestClass
{
    X = 10,
    Y = 20
};

var script = CSharpScript.Create<int>("X + Y", 
                                      ScriptOptions.Default,
                                      cls.GetType());
var func = script.CreateDelegate();
var res = func(cls).Result;
```

with following exception:
InvalidCastException: [A]TestClass cannot be cast to [B]TestClass. Type A originates from 'Logic, Version=2.2.0.0, Culture=neutral, PublicKeyToken=null' in the context 'Default' at location 'D:\Data\git\...\bin\Debug\netcoreapp1.1\Logic.dll'. Type B originates from 'Logic, Version=2.2.0.0, Culture=neutral, PublicKeyToken=null' in the context 'Default' at location 'D:\Data\git\...\bin\Debug\netcoreapp1.1\Logic.dll'.

**Expected Behavior**:

It should work :).

**Actual Behavior**:

After updating to RTM version of .net core 1.1.0 i cant run even simplest script, it always fails. with `InvalidCastException` for absolutely same type. 

I even tried to run `CSharpScript.EvaluateAsync`, but with same result.
I don`t know if this is bug here or in .net core 1.1.0, but definitely it is bug. 

This code was working with .net core 1.0.1.

Please i need to solve this so i can update to newest .net core. 
Thank you very much.

**Comment by kukjevov** [comment]
Sorry, but i need to know this, because we want to use some stuff from EFCore 1.1.0, but we can`t update because of scripting which is not working. 

Is there any estimate when will be this bug fixed ? 

Thank you :)

**Comment by tmat** [comment]
C# Scripting is currently only supported on .NET Framework 4.6+, not on .NET Core. We'll look into issues on .NET Core at some point but unfortunately I can't give any estimates on time frame.

**Comment by kukjevov** [comment]
Thank you for answer.

Well i will have to find workaround, but interesting thing is that in versions 1.0.0 and 1.0.1 of .NET Core it was working with some minor problems, but it was working. 

What is very strange that exception message says that it cant cast type to itself.

Thank you anyway :)

**Comment by robrisner** [comment]
I'm experiencing the same issue. Version 1.1.0 of .NET Core breaks C# Scripting code that works perfectly fine on version 1.0.1.

**Comment by tmat** [comment]
Fixed by #15658

**Comment by pongba** [comment]
@tmat is looks like the issue shows up in latest version again, see: https://github.com/dotnet/roslyn/issues/45197
